src/main/java/ai/datahunters/md/server/photos/upload/ArchiveHandler.java

```diff
@@ -0,0 +1,69 @@
+package ai.datahunters.md.server.photos.upload;
+
+import org.apache.commons.compress.archivers.ArchiveEntry;
+import org.apache.commons.compress.archivers.ArchiveInputStream;
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream;
+import org.apache.commons.compress.archivers.zip.ZipArchiveInputStream;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.tika.Tika;
+import org.apache.tika.metadata.Metadata;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import reactor.core.publisher.Mono;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
+public class ArchiveHandler {
+    public static final String TAR="application/x-gtar";
+    public static final String ZIP="application/zip";
+
+    private Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    public Mono<List<String>> unarchive(ArchiveInputStream archive) {
+        List<String> files = new ArrayList<>();
+        try {
+            ArchiveEntry entry;
+            logger.info(String.valueOf(archive.getBytesRead()));
+            while ((entry = archive.getNextEntry()) != null) {
+                if (entry.isDirectory()) {
+                    logger.info(entry.getName());
+//                    FileUtils.forceMkdir(new File(unarchivedDataDest.toUri()+entry.getName()));
+                } else {
+                    logger.info(entry.getName());
+                    IOUtils.copy(archive, FileUtils.openOutputStream(new File(entry.getName())));
+                }
+                files.add(entry.getName());
+            }
+
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return Mono.just(files);
+    }
+
+    public Mono<ArchiveInputStream> probeContent(InputStream in) {
+        Tika tika = new Tika();
+        try {
+            String datatype = tika.detect(in, new Metadata());
+            logger.info(datatype);
+            if (datatype.equals(ZIP)) {
+                return  Mono.just(new ZipArchiveInputStream(in));
+            } else if (datatype.equals(TAR)) {
+                return Mono.just(new TarArchiveInputStream(in));
```

Comment:
again, if you dont have anything useful for streams to do, its better to have a method which just do not mention stream on its signature